ops/testing.py

```diff
@@ -56,27 +61,140 @@
     storage,
 )
 
-# The Harness testing framework.
-_ = ActionFailed
-_ = ActionOutput
-_ = AppUnitOrName
-_ = CharmType
-_ = ExecArgs
-_ = ExecHandler
-_ = ExecResult
-_ = Harness
-_ = ReadableBuffer
-_ = YAMLStringOrFile
+# The Harness unit testing framework.
+__all__ = [
+    'ActionFailed',
+    'ActionOutput',
+    'AppUnitOrName',
+    'CharmType',
+    'ExecArgs',
+    'ExecHandler',
+    'ExecResult',
+    'Harness',
+    'ReadableBuffer',
+    'YAMLStringOrFile',
+]
+
+# If the 'ops.testing' optional extra is installed, make those
+# names available in this namespace.
+try:
+    _version = importlib.metadata.version('ops-scenario')
+except importlib.metadata.PackageNotFoundError:
+    pass
```

Comment:
Do we need `_version = None` here, so the `if _version` below doesn't raise a `NameError`?

Reply:
No, because the `if _version` is in the try's `else` block, so if the exception is raised it won't get to there.